app/response_processor.py

```diff
@@ -95,12 +95,25 @@ def send_receipt(self, decrypted_json):
                               ru_ref=decrypted_json['metadata']['ru_ref'])
             queue_ok = False
         elif decrypted_json.get("survey_id") == "census":
+            self.logger.info(
+                "About to publish receipt to ctp queue",
+                tx_id=decrypted_json['tx_id'],
```

Comment:
`tx_id` should already have been gleaned from the rabbit headers so you don't need to get it from the `decrypted_json` - it may already (and should be) be bound to the logger.